pkg/deploy/predeploy.go

```diff
@@ -405,6 +421,32 @@ func (d *deployer) configureServiceSecrets(ctx context.Context) error {
 	return d.ensureSecretKey(ctx, d.portalKeyvault, env.PortalServerSSHKeySecretName)
 }
 
+func (d *deployer) ensureAndRotateSecret(ctx context.Context, kv keyvault.Manager, secretName string, len int) error {
+	existingSecrets, err := kv.GetSecrets(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, secret := range existingSecrets {
+		if filepath.Base(*secret.ID) == secretName {
```

Comment:
The current implementation of `ensureSecret` [before this change does the same](https://github.com/Azure/ARO-RP/blob/master/pkg/deploy/predeploy.go#L408-L430)

This followed suit.  I believe the reasoning is because if we just call `kv.GetSecret()`, it will return an error if the secret doesn't exist, but also will return an error if the call fails for some reason.  

Fetching all the secrets in the keyvault (`kv.GetSecrets(ctx)`) and then getting the individual data out of the secret (`kv.GetSecret(ctx, secretName, "")`) ensures that we check for the existence of the secret rather than error check the first call, which the error is subject to change over time.